File: Core/src/main/java/jokatu/identity/IdentifiableReader.java
package jokatu.identity;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * An object that returns an {@link Identifiable} with the given {@link Identity}.
 * @author Steven Weston
 */
public interface IdentifiableReader<I extends Identity, J extends Identifiable<I>> {

	@Nullable
	J read(@NotNull I identity);
}

File: Core/src/main/kotlin/jokatu/identity/IdentifiableReader.kt
package jokatu.identity

/**
 * An object that returns an [Identifiable] with the given [Identity].
 * @author Steven Weston
 */
interface IdentifiableReader<I : Identity, J : Identifiable<I>> {

	/**
	 * @return the [Identifiable] with the given [Identity] or `null` if no such `Identifiable` exists
	 */
	fun read(identity: I): J?

	/**
	 * @return a count of all the entities that can be read
	 */
	fun count(): Long
}
